backend/src/middleware/validation/validationMiddleware.js

```diff
@@ -62,30 +64,33 @@ const validateReport = async (resolve, root, args, context, info) => {
   const { user, driver } = context
   if (resourceId === user.id) throw new Error('You cannot report yourself!')
   const session = driver.session()
-  const reportQueryRes = await session.run(
-    `
+  try {
+    const reportQueryRes = await session.run(
+      `
       MATCH (:User {id:$submitterId})-[:REPORTED]->(resource {id:$resourceId}) 
       RETURN labels(resource)[0] as label
     `,
-    {
-      resourceId,
-      submitterId: user.id,
-    },
-  )
-  session.close()
-  const [existingReportedResource] = reportQueryRes.records.map(record => {
-    return {
-      label: record.get('label'),
-    }
-  })
+      {
+        resourceId,
+        submitterId: user.id,
+      },
+    )
+    const [existingReportedResource] = reportQueryRes.records.map(record => {
+      return {
+        label: record.get('label'),
+      }
+    })
 
-  if (existingReportedResource) throw new Error(`${existingReportedResource.label}`)
-  return resolve(root, args, context, info)
+    if (existingReportedResource) throw new Error(`${existingReportedResource.label}`)
+    return resolve(root, args, context, info)
+  } finally {
+    session.close()
+  }
 }
 
 export default {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-03T13:03:53Z" title="Tuesday, December 3rd 2019, 2:03:53 pm +01:00">Dec 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -62,25 +65,28 @@ const validateReport = async (resolve, root, args, context, info) => {
   const { user, driver } = context
   if (resourceId === user.id) throw new Error('You cannot report yourself!')
   const session = driver.session()
-  const reportQueryRes = await session.run(
-    `
+  try {
+    const reportQueryRes = await session.run(
+      `
       MATCH (:User {id:$submitterId})-[:REPORTED]->(resource {id:$resourceId}) 
       RETURN labels(resource)[0] as label
     `,
-    {
-      resourceId,
-      submitterId: user.id,
-    },
-  )
-  session.close()
-  const [existingReportedResource] = reportQueryRes.records.map(record => {
-    return {
-      label: record.get('label'),
-    }
-  })
+      {
+        resourceId,
+        submitterId: user.id,
+      },
+    )
+    const [existingReportedResource] = reportQueryRes.records.map(record => {
+      return {
+        label: record.get('label'),
+      }
+    })
 
-  if (existingReportedResource) throw new Error(`${existingReportedResource.label}`)
-  return resolve(root, args, context, info)
+    if (existingReportedResource) throw new Error(`${existingReportedResource.label}`)
+    return resolve(root, args, context, info)
+  } finally {
+    session.close()
+  }
 }
 
 export default {
```


I'm sure these changes are fine, but it would make me feel a lot more if there were tests for this file, no?